server/src/utils.ts

```diff
@@ -135,17 +136,23 @@ async function validateOtherPageLinks(xmlData: Document, knownModules: Map<strin
     // A matching module was found. Also validate target ID if it's specified in the link
     const linkTargetId: string = linkElement.getAttribute('target-id')
     if (linkTargetId !== '') {
-      const targetModulePath = targetModuleInformation.path
-      const targetModuleText = fs.readFileSync(targetModulePath, 'utf-8')
-      const targetModuleXmlData: Document = new DOMParser().parseFromString(targetModuleText)
-      if (targetModuleXmlData === undefined) {
-        preparedDiagnostic.message = `Could not parse target document!: ${linkTargetModule}`
-        diagnostics.push(preparedDiagnostic)
-        continue
+      // Grab a previously parsed object if we've already checked something on
+      // this target page, otherwise read the file and parse it
+      let maybeModuleXmlData: Document | undefined = parsedOtherPages.get(linkTargetModule)
```

Comment:
Will this cache be kept in sync if I am editing the other page?

Reply:
Yes, so long as the event is fired to validate the first page. For example: If `m0001` has a link to `m00002`, and I modify the target ID / element `m0002` so the reference from `m0001` is bad, then the next time I open / edit `m0001` the error will show. 

Note: The need for the open / edit has nothing to do with the cache (similar to above it is purely intra-event), but with the events we currently hook into for validating a given document. So in terms of cache correctness it's the same as above in that the cache does not attempt to track changes across events.